semeio/workflows/localisation/local_script_lib.py

```diff
@@ -237,6 +239,257 @@ def activate_and_scale_correlation(
         logging.warning("Scaling method: %s is not implemented.", method)
 
 
+def apply_gaussian_decay(
+    row_scaling, data_size, grid, ref_pos, main_range, perp_range, azimuth
+):
+    row_scaling.assign(
+        data_size,
+        GaussianDecay(ref_pos, main_range, perp_range, azimuth, grid),
+    )
+
+
+def apply_exponential_decay(
+    row_scaling, data_size, grid, ref_pos, main_range, perp_range, azimuth
+):
+    row_scaling.assign(
+        data_size,
+        ExponentialDecay(ref_pos, main_range, perp_range, azimuth, grid),
+    )
+
+
+def apply_from_file(row_scaling, data_size, grid, filename, param_name, log_level):
+    debug_print(
+        f"Read scaling factors as parameter {param_name}", LogLevel.LEVEL3, log_level
+    )
+    debug_print(f"File name:  {filename}", LogLevel.LEVEL3, log_level)
+    with cwrap.open(filename, "r") as file:
+        scaling_parameter = Ecl3DKW.read_grdecl(
+            grid,
+            file,
+            param_name,
+            strict=True,
+            ecl_type=EclDataType.ECL_FLOAT,
+        )
+        for index in range(data_size):
+            global_index = grid.global_index(active_index=index)
+            row_scaling[index] = scaling_parameter[global_index]
+
+
+def get_min_max_region_number(region_parameter, missing_code):
+    # Get min and max region number
+    min_region_number = region_parameter[region_parameter != missing_code].min()
```

Comment:
This might not be relevant, but if this is a expensive operation you could do:
```suggestion
region_number = region_parameter[region_parameter != missing_code]
return region_number.min(), region_number.max()
```

Reply:
Fixed